lib/compat/wordpress-6.2/additional-filters.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * Filters used by Gutenberg to extend core functionality.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Filters the title of the default page template displayed in the drop-down on the post editor
+ *
+ * @param string $label   The display value for the default page template title.
+ * @param string $context Where the option label is displayed.
+ * @return string The new display value for the default page template title.
+ */
+function gutenberg_add_template_to_default_template_title( $label, $context = '' ) {
+	global $post_type, $post;
```

Comment:
Basically, if I'm understanding properly, we're trying to run WordPress's template resolution algorithm for the current global $post and $post_type. I feel this is not the first time I make this comment but it would be great if this algorithm could be abstracted in a dependency-less class/function. Is there a way to share code? (Probably not for this PR anyway)

Also, I want to ping @ockham here as I believe he worked on this kind of things previously for block templates resolution.

Also, this will run the template resolution for each editor page load, I'm assuming that's fast enough?

Reply:
> it would be great if this algorithm could be abstracted in a dependency-less class/function.

100%. It would be great to have something similar to [Hierarchy](https://github.com/Brain-WP/Hierarchy) in the core. A class that can resolve a template based on the current post or query context.

Reply:
> Basically, if I'm understanding properly, we're trying to run WordPress's template resolution algorithm for the current global $post and $post_type. I feel this is not the first time I make this comment but it would be great if this algorithm could be abstracted in a dependency-less class/function. Is there a way to share code? (Probably not for this PR anyway)

Sure, I agree that abstracting template resolution is a common need. However, it's not straightforward to implement. For example, in this case, we want template resolution to use the default/fallback template even if a custom template has been assigned. In other cases, we may need template resolution to work differently.

I don't think this should be a blocker for us. We can explore options for abstraction while also addressing the specific needs we have, and later consolidate.

> Also, this will run the template resolution for each editor page load, I'm assuming that's fast enough?

Yes, I think it is something fast enough we also call these functions on the site front end anyway, where performance is even more critical.